src/components/UserInputBlock.tsx

```diff
@@ -51,12 +66,66 @@ export const UserInputBlock = ({ content, prompt, ack, waitForEvent, fireEvent,
       .max(MAX_CHARACTER_INPUT_BLOCK_LENGTH, EXCEEDED_MAX_INPUT_ERROR)
   })
 
+  const updateInteractiveState = async (): Promise<void> => {
+    try {
+      if (contentId === undefined || persistor === undefined) {
+        setInteractiveState({
+          userResponse: '',
+          responseSubmitted: false
+        }
+        )
+        setPersistorStatus(PersistorStatus.Success)
+        return
+      }
+
+      const currentLocalStorageState = await persistor.get(contentId)
+      if (currentLocalStorageState !== null) {
+        const parsedLocalStorage = JSON.parse(currentLocalStorageState)
+        setInteractiveState({
+          userResponse: parsedLocalStorage.userResponse,
+          responseSubmitted: parsedLocalStorage.responseSubmitted
+        })
+        setResponseSubmitted(parsedLocalStorage.responseSubmitted)
+      } else {
+        setInteractiveState({
+          userResponse: '',
+          responseSubmitted: false
+        })
+      }
+      setPersistorStatus(PersistorStatus.Success)
+    } catch (err) {
+      setPersistorStatus(PersistorStatus.Failure)
+      console.error(err)
+    }
+  }
+
+  useEffect(() => {
+    updateInteractiveState().catch(() => { })
+  }, [])
+
   const handleSubmit = async (values: InputFormValues, { setFieldError }: FormikHelpers<InputFormValues>): Promise<void> => {
     if (values.response.trim() === '') {
       setFieldError('response', NON_EMPTY_VALUE_ERROR)
       return
     }
 
+    try {
+      if (contentId !== undefined && persistor !== undefined) {
+        await persistor.put(contentId, JSON.stringify({ userResponse: values.response, responseSubmitted: true }))
+        const currentLocalStorageState = await persistor.get(contentId)
+        if (currentLocalStorageState !== null) {
+          const parsedLocalStorage = JSON.parse(currentLocalStorageState)
+          setInteractiveState({
+            userResponse: parsedLocalStorage.userResponse,
+            responseSubmitted: parsedLocalStorage.responseSubmitted
+          })
+        }
+      }
+    } catch (error) {
+      setPersistorStatus(PersistorStatus.Failure)
```

Comment:
I'm not sure this is the best way to handle the scenario where a `.put()` fails since it would effectively hide everything and the user can't proceed in any reasonable way. What about something like the following instead where we effectively treat being able to save the response as post-submit validation?:

```javascript
      setFieldError('response', 'Error saving response. Please try again.')
      return
```

Reply:
Fully acknowledge it was not the best way to handle the failure of the `.put()`. Because this is a spike I did not put too much thought into error handling, but your implementation is simple enough that I'll keep it in mind moving forward. Thanks!